src/components/cards/ValidationList/ValidationList.tsx

```diff
@@ -0,0 +1,112 @@
+import { useEffect, useState } from 'react';
+
+import { ValidationObject } from '../../../types';
+import { AlertCircle, CircleCheck } from '../../icons';
+import { Tooltip } from '../../layout';
+
+function ValidationList({ validations }: { validations: ValidationObject[] }) {
+  const [validationChecklist, setValidationChecklist] =
+    useState<Array<JSX.Element | undefined>>();
+
+  useEffect(() => {
+    if (validations) {
+      updateValidationList(validations);
+    }
+  }, [validations]);
+
+  function updateValidationList(validations: ValidationObject[]) {
+    const newElements = validations.map(
+      (validationItem: ValidationObject, index: number) => {
+        console.log(validationItem.status);
+
+        switch (validationItem.status) {
+          case true: {
+            return (
+              <>
+                <Tooltip key={index} message={'Success!'}>
+                  <span className="flex flex-row text flex-left white bold">
+                    <CircleCheck
+                      width={20}
+                      height={20}
+                      fill={'var(--success-green)'}
+                    />
+                    {validationItem.name}
+                    <AlertCircle
+                      width={'16px'}
+                      height={'16px'}
+                      fill={'var(--text-white)'}
+                    />
+                  </span>
+                </Tooltip>
+              </>
+            );
+          }
+
+          case false: {
+            return (
+              <>
+                <Tooltip key={index} message={validationItem.error}>
+                  <span className="flex flex-row text flex-left white bold">
+                    <AlertCircle
+                      width={20}
+                      height={20}
+                      fill={'var(--error-red)'}
+                    />
+                    {validationItem.name}
+                    <AlertCircle
+                      width={'16px'}
+                      height={'16px'}
+                      fill={'var(--text-white)'}
+                    />
+                  </span>
+                </Tooltip>
+              </>
+            );
+          }
+          default:
+            return (
+              <>
+                <Tooltip key={index} message={validationItem.status}>
+                  <span className="flex flex-row text flex-left white bold">
+                    <AlertCircle
+                      width={20}
+                      height={20}
+                      fill={'var(--text-faded)'}
+                    />
+                    Invalid Status Code
+                    <AlertCircle
+                      width={'16px'}
+                      height={'16px'}
+                      fill={'var(--text-white)'}
+                    />
```

Comment:
do we need the switch statment? it seems the only difference are the text and fill, can we do a ternary operator on those so this is less redundant?

Reply:
Right now we dont, but, if we want to upgrade to a more advanced validation display in the future (like initial implementation with loading states) then I think a switch statement is the way to go.

Reply:
maybe, but it's beefy and bloated for now. i'd prefer we create the proper object and then do the component rendering in JSX